src/json.cr

```diff
@@ -63,11 +63,15 @@ module JSON
 
   # Exception thrown on a JSON parse error.
   class ParseException < Error
-    getter line_number : Int32
-    getter column_number : Int32
+    getter line_number : Int32?
+    getter column_number : Int32?
 
-    def initialize(message, @line_number, @column_number)
-      super "#{message} at #{@line_number}:#{@column_number}"
+    def initialize(message, @line_number = nil, @column_number = nil)
+      if @line_number && @column_number
+        super "#{message} at #{@line_number}:#{@column_number}"
+      else
+        super "#{message}"
+      end
```

Comment:
Explanation why this condition is needed:

After `JSON::ParseException` is rescued, the exception class will be initialized again with original exception message:  https://github.com/crystal-lang/crystal/pull/4873/files#diff-3f54210a12cee69b8bbb656af3626edfR124

This would create an exception message like:
```
Error parsing attribute JSONPerson.name: Expected string but was int at 2:13 at 2:13
```

After the exception is rescued, the location should not be set again(otherwise `location` appears twice like above). So I made `@line_number` and `@column_number` nilable.

Reply:
I don't like this. `line_number` and `column_number` aren't *just* for humans. They are for machines too and should be set properly. I would introduce an option to the constructor which skips appending these values.

Reply:
See https://github.com/crystal-lang/crystal/pull/4873#discussion_r135506172. I believe next commit changes this.

Reply:
It's not fixed currently?

Reply:
Unfortunately... I got response, but I didn't see such a change. @kenta-s said "I will try to fix that.", so I'm waiting.